cli/cli/commands/run/run.go

```diff
@@ -551,17 +552,21 @@ func getOrCreateEnclaveContext(
 	return enclaveContext, isNewEnclaveFlagWhenCreated, nil
 }
 
-// validatePackageArgs just validates the args is a valid JSON string
+// validatePackageArgs just validates the args is a valid JSON or YAML string
 func validatePackageArgs(_ context.Context, _ *flags.ParsedFlags, args *args.ParsedArgs) error {
-	serializedJsonArgs, err := args.GetNonGreedyArg(inputArgsArgKey)
+	serializedArgs, err := args.GetNonGreedyArg(inputArgsArgKey)
 	if err != nil {
 		return stacktrace.Propagate(err, "An error occurred getting the script/package arguments using flag key '%v'", inputArgsArgKey)
 	}
 	var result interface{}
-	if err := json.Unmarshal([]byte(serializedJsonArgs), &result); err != nil {
-		return stacktrace.Propagate(err, "Error validating args, likely because it is not a valid JSON.")
+	var jsonError error
+	if jsonError = json.Unmarshal([]byte(serializedArgs), &result); jsonError == nil {
+		return nil
 	}
-	return nil
+	if err := yaml.Unmarshal([]byte(serializedArgs), &result); err == nil {
+		return nil
+	}
+	return stacktrace.Propagate(jsonError, "Error validating args, likely because it is not a valid JSON or YAML.")
```

Comment:
it's propagating `jsonError` but what happens if `err is also. != nil` could be confusing for users if both are != nil because we will printing only the error for the first one.